jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -144,10 +144,20 @@ private Decoder createDecoder(ObjectMapper objectMapper) {
 
     private feign.Client createOkHttpClient(String userAgent) {
         okhttp3.OkHttpClient.Builder client = new okhttp3.OkHttpClient.Builder();
+
+        // SSL
         if (config.getSslSocketFactory().isPresent()) {
-            client.sslSocketFactory(config.getSslSocketFactory().get());
+            Preconditions.checkArgument(config.getX509TrustManager().isPresent(),
```

Comment:
it's split-brained in this repo, but more places don't import Preconditions. I'll leave it as it is to stay with the majority.
````
rfink@rfink-mbp:~/git/http-remoting (rfink/retrofit2-api)$ f "*.java" | xargs grep " checkArgument" | wc -l
       9
rfink@rfink-mbp:~/git/http-remoting (rfink/retrofit2-api)$ f "*.java" | xargs grep "Preconditions.checkArgument" | wc -l
      27
rfink@rfink-mbp:~/git/http-remoting (rfink/retrofit2-api)$
````